runtime/compiler/control/CompilationThread.cpp

```diff
@@ -7519,7 +7519,18 @@ TR::CompilationInfoPerThreadBase::preCompilationTasks(J9VMThread * vmThread,
          }
       }
 #if defined(J9VM_OPT_JITSERVER)
-   bool cannotDoRemoteCompilation = cannotPerformRemoteComp(
+   TR_Debug *debug = TR::Options::getDebug();
+   bool dontRemoteCompile = false;
+   if(debug)
+   {
+      setCompilationShouldBeInterrupted(0); // zero the flag because createResolvedMethod can
+					   // acquire/releaseVMaccessIfNeeded and may see the flag set by previous compilation
+      TR_ResolvedMethod *resolvedMethod = fe->createResolvedMethod(&trMemory, (TR_OpaqueMethodBlock *)method);
+      dontRemoteCompile = !_compInfo.methodCanBeRemotelyCompiled(resolvedMethod->signature(&trMemory),resolvedMethod->convertToMethod()->methodType());
+      if (TR::Options::getVerboseOption(TR_VerboseJITServer) && dontRemoteCompile)
+         TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer, "Do local compilation for method=%s as specified in the remoteCompileExclude option",resolvedMethod->signature(&trMemory));
+   }
+   bool cannotDoRemoteCompilation = dontRemoteCompile || cannotPerformRemoteComp(
```

Comment:
I wonder if the code flows better like this:
```
 bool cannotDoRemoteCompilation = cannotPerformRemoteComp(....)
 if (TR::Options::getDebug() && !cannotDoRemoteCompilation  && _compInfo.getPersistentInfo()->getRemoteCompilationMode() == JITServer::CLIENT)
   {
   setCompilationShouldBeInterrupted(0); // zero the flag because createResolvedMethod can
					 // acquire/releaseVMaccessIfNeeded and may see the flag set by previous compilation
   TR_ResolvedMethod *resolvedMethod = fe->createResolvedMethod(&trMemory, (TR_OpaqueMethodBlock *)method);
   cannotDoRemoteCompilation = !_compInfo.methodCanBeRemotelyCompiled(resolvedMethod->signature(&trMemory),resolvedMethod->convertToMethod()->methodType());
   }